src/InsetOrderOptimizer.cpp

```diff
@@ -351,6 +355,8 @@ void InsetOrderOptimizer::processOuterWallInsets()
                 if (extruder_nr == mesh.getSettingAsExtruderNr("wall_0_extruder_nr"))
                 {
                     gcode_layer.addPolygon(*inset_polys[0][0], outer_poly_start_idx, mesh_config.inset0_config, wall_overlapper_0, wall_0_wipe_dist, spiralize, flow, retract_before_outer_wall);
+                    // move inside so an immediately following retract doesn't occur on the outer wall
+                    moveInside(wall_line_width_x * num_insets);
```

Comment:
Ideally you'd use the line width in the `mesh_config.insetX_config` object, in case that is modified by the initial layer line width.